src/uwtools/scheduler.py

```diff
@@ -35,13 +37,15 @@ def directives(self) -> list[str]:
         pre, sep = self._prefix, self._directive_separator
         ds = []
         for key, value in self._processed_props.items():
+            if key in self._forbidden_directives:
+                msg = "Directive '%s' invalid for scheduler '%s'"
+                raise UWConfigError(msg % (key, self._scheduler))
             if key in self._managed_directives:
                 switch = self._managed_directives[key]
-                ds.append(
-                    "%s %s" % (pre, switch(value))
-                    if callable(switch)
-                    else "%s %s%s%s" % (pre, switch, sep, value)
-                )
+                if callable(switch) and (x := switch(value)) is not None:
+                    ds.append("%s %s" % (pre, x))
+                else:
+                    ds.append("%s %s%s%s" % (pre, switch, sep, value))
```

Comment:
Some scheduler switches, like `--verbose`, have no value: They're booleans and are true if present and false if absent. So, only include the switch at all if its callable, which normally returns the value to associate with the switch, returns a value other than `None`. See below for the Slurm implementations of these callables.